### DISCUSSION TITLE
Way of circuit breaker implementation in common webclient class

### DISCUSSION THREAD

**Discussion by lukhong** [resilience4j/resilience4j#2014] [Q&A] [answered]
Hi, team. 
I want to implement circuit breaker with resilience4 to my microservice.
Let me introduce the architecture simply.

**Request Server ----(1)----> My Server ----(2)----> Various External Servers**

Example of (1) 
- GET /company_a/users
- GET /company_a/emails
- POST /company_a/contents

- GET /company_b/users
- GET /company_b/emails
- POST /company_b/contents

...

Example of (2) to each endpoint URL
- GET /users
- GET /emails
- POST /contents

My server gets request from `Request Server` and then call external service. 
External services are many and the hosts are different of course. But they all gets same request with the same spec. 
So the `webclient` of My Server could be common and provides methods for /users, /emails, /contents. 
What we should do for the request-(1) is just replacing the target URL. It is very good to handle increasing external services.

But when we consider the circuit breaker, I lost the way. 
If the continuous errors are coming from GET /company_b/users, then I should open only for external service of company b. 
But I have a common webclient for /users httprequest. Is there a way to implement resilience4j circuit breaker pattern to this case?
I am using spring boot JAVA

**Comment by RobWin** [comment, marked as answer]
Ok,
so the different external services are hosted and managed by different companies and thus have different failure domains, correct?
Usually I would recommend that you have a dedicated circuitbreaker instance per company. How do you load balance the traffic to different external services? 
You could make use of Spring Expression Language to pick the correct circuitbreaker by using a method parameter. 

```
@CircuitBreaker(name = "#root.args[0]")
    public String execute(String targetIdentifier) {
        return ...
    }
```

**Reply by lukhong** [reply]
Thank you @RobWin . It works for me.
